lib/python/pypsrcat.py

```diff
@@ -65,13 +63,13 @@ def __init__(self, line):
                         d, m = dms
                         s = 0.0
                     elif len(dms)==1:
-                        d = dms
+                        d = dms[0]
                         m = s = 0.0
 		    # Fixed silly dec=-0.0 bug
-		    if parts[part_index].split(":")[0]=="-00":
-			m = -m
-			s = -s
-		    self.dec = psr_utils.dms_to_rad(d, m, s)
+                    if parts[part_index].split(":")[0]=="-00":
```

Comment:
I strongly recommend that parsing angle strings like this be done in astropy, so that dec=-0 bugs and the like don't creep in.

Reply:
Yup.  That is old code that I have already flagged to convert.  I'm going to update the way that the ATNF pulsar data are imported as well.  Just haven't had time to do it yet.

Reply:
Yeah, I figured. I just like to pester you :-)

Reply:
I don't blame you!  I'd pester me too!  That code is atrocious!!!